arch/platform/cooja/sys/cooja_mt.c

```diff
@@ -36,36 +36,95 @@
  * Fredrik Osterlind <fros@sics.se>
  */
 
-#include "contiki.h"
+#ifdef __APPLE__
+#define _DARWIN_C_SOURCE /* For pthread_get_stackaddr_np() */
+#else
+#define _GNU_SOURCE /* For pthread_getattr_np(). */
+#endif
+#include <pthread.h>
+
 #include "sys/cooja_mt.h"
-#include "sys/cc.h"
+
+#include <signal.h>
+#include <stdio.h>
 
 #define MT_STATE_READY   1
 #define MT_STATE_RUNNING 2
 #define MT_STATE_EXITED  5
 
 static struct cooja_mt_thread *current;
+static struct cooja_mt_thread *cooja_running_thread;
```

Comment:
What is the difference between `current` and `cooja_running_thread`? Are both needed?

Reply:
The old code had both, and I'm not sure why. They are not always the same value, so I left both so the PR wouldn't introduce new behaviors.

Reply:
I think the reason was that the variables were in different files and local in each file. The variables have now moved to the same file and seem to serve the same purpose, but let us leave it for another PR if you like.